tests/contrib/futures/test_propagation.py

```diff
@@ -293,10 +293,6 @@ def fn():
                 future = executor.submit(fn)
                 time.sleep(0.01)
 
-        # assert main thread span is finished first
-        self.assert_span_count(1)
-        self.assert_structure(dict(name="main.thread"))
```

Comment:
This is no longer applies because the trace is kept until all spans have finished. Before, the futures spans would be per-future so the main thread span would be flushed independently of the child span